packages/utils/src/date/Locale.ts

```diff
@@ -0,0 +1,556 @@
+      const hourPart = parts.find(part => part.type === 'hour');
+      if (hourPart && dayPeriodPart) {
+        periodNames.push({
+          value: Number(hourPart.value),
+          name: dayPeriodPart.value.toLowerCase()
+        });
+      }
+    }
+    return periodNames;
+  }
+
+  /**
+   * Get period names for the specified formatter
+   * @returns names Period names
+   */
+  private formatEraNames (): FormattedNames[] {
+    const eraNames: FormattedNames[] = [];
+    const dates = [utcParse('-0001-01-01').getTime(), Date.UTC(2018, 0, 1)];
+    for (let i = 0; i <= 1; i += 1) {
+      const era = dates[i];
+      const part = this.formatter.formatToParts(era).find(part => part.type === 'era');
+      if (part) {
+        eraNames.push({
+          value: era,
+          name: part.value.toLowerCase()
+        });
+      }
+    }
+    return eraNames;
+  }
+
+  /**
+   * Escape list with literals to be used with RegExp
+   * @param list A list with items
+   * @returns escaped list
+   */
+  private escapeList (list: FormattedNames[]): string {
+    // There might be duplicates, which are not required in RegExp
+    return `(${[...new Set(list.map(item => item.name))].map(item => regExpEscape(item)).join('|')})`;
+  }
+
+  private _regExp: RegExp | null = null;
+  /**
+   * Get regular expression for the format
+   */
+  private get regExp (): RegExp {
+    if (this._regExp) {
+      return this._regExp;
+    }
+    const options = this.options;
+    const parts = this.parts;
+
+    let str = '';
+
+    for (let i = 0; i < parts.length; i += 1) {
+      const part = parts[i];
+      switch (part.type) {
+        case 'year':
+          str += options.year === TwoDigit ? YY : YYYY;
+          break;
+        case 'month':
+          str += options.month === TwoDigit || options.month === Numeric ? MM : this.escapeList(this.monthNames);
+          break;
+        case 'day':
+          str += DD;
+          break;
+        case 'hour':
+          const hourCycle = this.hourCycle;
+          str += hourCycle === 'h23' ? H23 : hourCycle === 'h12' ? H12 : hourCycle === 'h11' ? H11 : H24;
+          break;
+        case 'minute':
+          str += mm;
+          break;
+        case 'second':
+          str += ss;
+          break;
+        case 'fractionalSecond':
+          const ms = this.fractionalSecondDigits;
+          str += ms === 1 ? S : ms === 2 ? SS : SSS;
+          break;
+        case 'weekday':
+          str += this.escapeList(this.weekdayNames);
+          break;
+        case 'dayPeriod':
+          str += this.escapeList(this.periodNames);
+          break;
+        case 'literal':
+          str += `(${regExpEscape(part.value.toLowerCase())})`;
+          break;
+        case 'era':
+          str += this.escapeList(this.eraNames);
+          break;
+        case 'timeZoneName':
+          throw new Error('Parsing timezones is ambiguous. Make sure that `timeZoneName` option is not provided');
+          break;
```

Comment:
This `break` is unreachable. Ref https://sonarcloud.io/project/issues?resolved=false&types=BUG&pullRequest=234&id=refinitiv_refinitiv-ui&open=AX-2U44Mz09b4xyDNThu